src/util/subscription-set/subscription-set.ts

```diff
@@ -9,27 +9,31 @@ import { RedisSubscriptionSet } from './redis-sorted-set'
  */
 export interface SubscriptionSet<T> {
   /** Add a new subscription to the set */
-  add(value: T, ttl: number): PromiseOrValue<void>
+  add(value: T, ttl: number, key?: string): PromiseOrValue<void>
```

Comment:
Since you need the key for the expiring sorted set, I think you'll need to make a change in the abstract subscription transport to pass the cache key when we add to the subscription set

Reply:
https://github.com/smartcontractkit/ea-framework-js/blob/063e3bb44a835871b7744c02e7dc9cc10c64cf2e/src/transports/abstract/subscription.ts#L44